plugin/src/main/java/io/jenkins/plugins/analysis/warnings/groovy/ParserConfiguration.java

```diff
@@ -84,6 +87,50 @@ public void setParsers(final List<GroovyParser> parsers) {
         save();
     }
 
+    /**
+     * Says if the admin has permitted groovy parsers to scan a build's console log.
+     *
+     * @return true if groovy parsers can scan the console, false if they are
+     *         limited to only scanning files on the build node.
+     */
+    public boolean isConsoleLogScanningPermitted() {
+        return consoleLogScanningPermitted;
+    }
+
+    /**
+     * Sets whether or not the admin has permitted groovy parsers to scan a build's
+     * console log.
+     *
+     * @param consoleLogScanningPermitted true if groovy parsers can scan the
+     *                                    console, false if they are limited to only
+     *                                    scanning files on the build node.
+     */
+    @DataBoundSetter
+    public void setConsoleLogScanningPermitted(final boolean consoleLogScanningPermitted) {
+        this.consoleLogScanningPermitted = consoleLogScanningPermitted;
+
+        save();
+    }
+
+    /**
+     * Called by jelly to validate the configured value that could be passed to
+     * {@link #setConsoleLogScanningPermitted(boolean)}.
+     * 
+     * @param value The current value.
+     * @return {@link FormValidation#ok()} if all is well, else a warning or error.
+     */
+    // Maintenance note: This is NOT annotated with @POST because it does not leak secrets.
+    // Similarly, there's no need to permission-check either.
+    // If it is ever changed to do anything non-trivial that might leak secrets then
+    // it must be changed to @POST and checkMethod="post" be added to the checkbox.
+    // There is a matching entry in archunit_ignore_patterns.txt to permit this.
```

Comment:
Note: I did try adding checkMethod="post" to the checkbox element, but it didn't work - it looks like checkbox does not support that attribute (and there's no safe-checkbox either).
i.e. right now, we have no other option but to use a non-POST callback ... but that's OK because this doCheck method does not leak data (you either get a fxied-text warning, or you get no error - there's no side-effects).